board/app/models/user.php

```diff
@@ -56,39 +64,70 @@ class User extends AppModel {
 
         'password' => array(
             'length' => array(
-                'validate_between', self::MIN_PASSWORD_LENGTH, self::MAX_PASSWORD_LENGTH,
+                'validate_between', self::MIN_PASSWORD_LENGTH, self::MAX_PASSWORD_LENGTH
             )
         ),
         'confirm_password' => array(
             'match' => array(
                 'is_password_match',
-                )
-        ),
+            )
+        )
     );
 
+    /*
+    * Limits a certain field into a specific number of characters.
+    * Checks if the characters are valid. (e.g., dot, hyphen)
+    */
+    public function validate_username($username)
+    {
+        $valid = array('-', '_', '.');    
+        return ctype_alnum(str_replace($valid, '', $username));
+    }
+
+    public function validate_name($string)
+    {
+        $valid = array('-', ' ');    
+        return ctype_alpha(str_replace($valid, '', $string));
```

Comment:
The php built-in functions ctype_alnum & ctype_alpha returns a boolean value. Therefore you must rename the function names for validating the username and name po :bow: